add tidbcloud serverless driver node example (#15147)
pingcap/docs: tidb-cloud/serverless-driver-node-example.md

- A TiDB Serverless cluster. If you don't have any, you can [create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md)

## Create the project

1. Create a project named `node-example`:

    ```
    mkdir node-example
    cd node-example
    ```

2. Run the following command to install the driver with your preferred package manager, here we use npm as an example. This should create a `node_modules` directory and a `package-lock.json` file in your project directory.

    ```
    npm install @tidbcloud/serverless
    ```

## Use the serverless driver

> The serverless driver supports both the CommonJS and ESM module. We use ESM module as an example in this tutorial.

1. First you need to obtain the connection string for your database from the **Connect window** on TiDB Cloud cluster overview page. Your TiDB Serverless connection string will look something like this:

    ```
   mysql://[username]:[password]@[host]/[database]
    ```
   
2. Then add `type: "module"` to your `package.json` to specify ESM module. The `package.json` should look like this:

> 

Reviewer: 前面安装部分提到的是 a `package-lock.json` file in your project directory，但这里提到的是  `package.json` file，需要一致吗

Author: 我一致了

---
title: TiDB Cloud Serverless Driver Node.js Tutorial
summary: Learn how to use TiDB Cloud serverless driver in a local Node.js project.
---

# TiDB Cloud Serverless Driver Node.js Tutorial

> Learn how to use TiDB Cloud serverless driver in a local Node.js project.

Reviewer: ```suggestion
This tutorial describes how to use TiDB Cloud serverless driver in a local Node.js project.
```

> - To learn how to use TiDB Cloud serverless driver with Cloudflare Workers, Vercel Edge Functions, and Netlify Edge Functions, check out our [Insights into Automotive Sales](https://car-sales-insight.vercel.app/) and the [sample repository](https://github.com/tidbcloud/car-sales-insight).

## Before you begin

To complete this step-by-step tutorial, you need the following:

- The [Node.js](https://nodejs.org/en) >= 18.0.0.
- The [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) or your preferred package manager.
- A TiDB Serverless cluster. If you don't have any, you can [create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md).

## Step 1. Create a local Node.js project

1. Create a project named `node-example`:

    ```
    mkdir node-example
    cd node-example
    ```

2. Install the TiDB Cloud serverless driver using npm or your preferred package manager.

    The following command takes installation with npm as an example. Executing this command will create a `node_modules` directory and a `package.json` file in your project directory.

    ```
    npm install @tidbcloud/serverless
    ```

## Step 2. Use the serverless driver

> The serverless driver supports both CommonJS and ESM modules. The following steps take the usage of the ESM module as an example.

Reviewer 2: A bit weird to say "ESM Module" since ESM = **E**CMA**S**cript **M**odule. It's either **ESM** or **ES Module**.
```suggestion
The serverless driver supports both CommonJS and ES modules. The following steps take the usage of the ES module as an example.
```

Author: ESM means ES modules I think.

> 1. On the overview page of your TiDB Cloud Serverless cluster, click **Connect** in the upper-right corner, and then get the connection string for your database from the displayed dialog. The connection string looks something like this:

Reviewer 2: ```suggestion
1. On the overview page of your TiDB Cloud Serverless cluster, click **Connect** in the upper-right corner, and then get the connection string for your database from the displayed dialog. The connection string looks like this:
```

    ```
   mysql://[username]:[password]@[host]/[database]
    ```
   

> 2. In the `package.json` file, specify the ESM module by adding `type: "module"`. 

Reviewer 2: ```suggestion
2. In the `package.json` file, specify the ES module by adding `type: "module"`. 
```

Author: ditto

    For example:

>    ```json
>    {
>      "type": "module",
>      "dependencies": {
>        "@tidbcloud/serverless": "^0.0.7",
>      }
>    }
>    ```

Reviewer 2: ```suggestion
    ```json
    {
      "type": "module",
      "dependencies": {
        "@tidbcloud/serverless": "^0.0.7",
      }
    }
    ```
```

3. Create a file named `index.js` in your project directory and add the following code:

    ```js
    import { connect } from '@tidbcloud/serverless'
    
    const conn = connect({url: 'mysql://[username]:[password]@[host]/[database]'}) // replace with your TiDB Serverless cluster information
    console.log(await conn.execute("show tables"))
    ```

4. Run your project with the following command:

    ```
    node index.js
    ```

> ## Compatability with Node.js earlier versions

Reviewer 2: ```suggestion
## Compatability with earlier versions of Node.js
```

> If you are using Node.js earlier than 18.0.0, which does not have a global `fetch`, you can take the following steps to get `fetch`:

Reviewer 2: ```suggestion
If you are using Node.js earlier than 18.0.0, which does not have a global `fetch` function, you can take the following steps to get `fetch`:
```

---
title: Use TiDB Cloud Serverless Driver (Beta) in Node.js
summary: Learn how to use TiDB Cloud Serverless Driver in Node.js.
---

# Use TiDB Cloud Serverless Driver in Node.js

Learn how to use TiDB Cloud Serverless Driver in Node.js.

## Before you begin

To complete this step-by-step tutorial, you need:

- [Node.js](https://nodejs.org/en) >= 18.0.0.
- [NPM](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) or your preferred package manager.
- A TiDB Serverless cluster. If you don't have any, you can [create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md)

> ## Build the project

Reviewer 3: ```suggestion
## Create the project
```
